draft-ietf-mmusic-4572-update.xml

```diff
@@ -144,13 +144,15 @@ NEW TEXT:
     it is not known which certificate will be used when the
     fingerprints are exchanged. In such cases, one or more fingerprints
     MUST be calculated for each possible certificate. An endpoint
-    MUST, as a minimum, calculate a fingerprint using the 'SHA-256'
-    hash function algorithm for each possible certificate, unless the
-    endpoint knows that the peer supports a stronger algorithm, or
-    unless the endpoint knows that the peer has not been upgraded to
-    support the 'SHA-256' algorithm, or unless the endpoint is used for
-    a service, or within an environment that mandates usage of a
-    stronger algorithm.
+    MUST, as a minimum, calculate a fingerprint using both the 'SHA-256'
+    hash function algorithm and the hash function used to generate the
+    signature on the certificate for each possible certificate.
+    Including the hash from the signature algorithm ensures
+    interoperability with strict implementations of RFC 4572.
+    Either of these fingerprints MAY be omitted if the endpoint includes
+    a hash with a stronger hash algorithm that it knows that the peer
+    supports, if it is known that the peer does not support the hash
+    algorithm, or if local policy mandates use of stronger algorithms.
```

Comment:
This suppose to mean that local security policy can overwrite in cases of closed communication networks where all end points are known to support something more secure or only support something different. I am actually more concerned with SHA-256 here then the certificate signature. Certificate signature on self-signed certificate can always be made to match the fingerprint function you want.